### ISSUE TITLE
Do not dispose inner stream

### ISSUE THREAD

**Issue by ErikBogg** [MiloszKrajewski/lz4net#3]
Hi,

You should not dispose the inner stream when disposing the LZ4Stream. For two resons:
1) You are not the owner of the stream. It is created elsewhere, so it should be disposed elsewhere.
2) It is acceptable when using FileStream, but when using MemoryStream, it is totally worthless...

I'm using the LZ4Stream like this:

    public void Compress(Stream sIn, Stream sOut) {
        using (LZ4Stream lz4s = new LZ4Stream(sOutput , LZ4StreamMode.Compress)) {
            fsInput.CopyTo(lz4s);
        }
    }

The easiest way would be to just remove line 452 in [LZ4Stream.cs](https://github.com/MiloszKrajewski/lz4net/blob/master/src/LZ4/LZ4Stream.cs#L452). But you should perhaps make a thought to be backward compatable.

Kind regards
Erik

**Comment by MiloszKrajewski** [comment]
I do this because that's what DefalteStream does. If you learnt how to code around DeflateStream you can use same "trick" here. Personally I have very thin "IsolatingStream" which can wrap any other stream to protect it from closing. With MemoryStream you can use combination Flush with ToByteArray to get bytes. See implementation of LZ4Codec.Wrap(...) for details.

I can add doNotCloseInnerStream arguement to make it easier though, but there is nothing current implementation prevents, maybe is just a little bit tedious.

**Comment by ErikBogg** [comment]
Well, my "trick" would be to change:

    using (var deflate = new DeflateStream(stream, CompressionMode.Compress))

into:

    using (var deflate = new DeflateStream(stream, CompressionMode.Compress, true))

According to [MSDN](https://msdn.microsoft.com/en-us/library/system.io.compression.deflatestream(v=vs.100).aspx#Anchor_2)

**Comment by MiloszKrajewski** [comment]
```c#
using (var m = new MemoryStream())
using (var z = new LZ4Stream(m))
{
    //...
    z.Flush();
    return m.ToByteArray();
}
````

but you are right, it the constructor with this boolean argument should be added.
Soon.

**Comment by MiloszKrajewski** [comment]
FYI: has been addressed but neither tested or relased

**Comment by MiloszKrajewski** [comment]
Fixed in 1.0.10.93, see LZ4StreamFlags.IsolateInnerStream
